### DISCUSSION TITLE
Learn NextJS - Chapter 14: Improving Accessibility

### DISCUSSION THREAD

**Discussion by easoftwareGit** [vercel/next.js#58810] [Help]
### Summary

I have completed Chapter 14 twice, but I cannot get the custom error messages for data input to work. I have double checked my code to the examples. Is there anything that I need to add to my code that is not in the examples?

### Example

https://nextjs.org/learn/dashboard-app/improving-accessibility

**Comment by nicitaacom** [comment]
Describe your issue in details which result you await
Which result do you have now
Which code you use to achieve desired result
Also preffered is the main goal why do you want to get that result

**Comment by saifion33** [comment]
@easoftwareGit I also encounter same issue , because documentation is wrong for showing error.

here you can see
![Screenshot 2023-11-23 at 20-47-49 Learn Next js Improving Accessibility Next js](https://github.com/vercel/next.js/assets/100059226/80e134a7-4bf3-47d3-906d-6b74e71a88d6)

in line no 4 and 10 use required_error instead of invalid_type_error 
like this 
![Screenshot 2023-11-23 205204](https://github.com/vercel/next.js/assets/100059226/f7e4f3de-b43a-4af6-a091-dca756f0bed7)

**Reply by benibangun** [reply]
> @easoftwareGit I also encounter same issue , because documentation is wrong for showing error.
> 
> here you can see ![Screenshot 2023-11-23 at 20-47-49 Learn Next js Improving Accessibility Next js](https://private-user-images.githubusercontent.com/100059226/285249706-80e134a7-4bf3-47d3-906d-6b74e71a88d6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzkwMTQ1MjksIm5iZiI6MTczOTAxNDIyOSwicGF0aCI6Ii8xMDAwNTkyMjYvMjg1MjQ5NzA2LTgwZTEzNGE3LTRiZjMtNDdkMy05MDZkLTZiNzRlNzFhODhkNi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIwOFQxMTMwMjlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lODE0ZDcyMTcxYjhhMzY0OWRhZWE3MjhmYmQ2MjBiNTJjYTVmMjc5ODVmNWY2YmRiMmMwNTY1MDAxNWY0MWVkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.85HxfiB5Ieg3qFDXhe-g_eaYuGtrG4GaZDlHJQgIQ6U)
> 
> in line no 4 and 10 use required_error instead of invalid_type_error like this ![Screenshot 2023-11-23 205204](https://private-user-images.githubusercontent.com/100059226/285250655-f7e4f3de-b43a-4af6-a091-dca756f0bed7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzkwMTQ1MjksIm5iZiI6MTczOTAxNDIyOSwicGF0aCI6Ii8xMDAwNTkyMjYvMjg1MjUwNjU1LWY3ZTRmM2RlLWI0M2EtNGFmNi1hMDkxLWRjYTc1NmYwYmVkNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIwOFQxMTMwMjlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05MjUwNjlkODA4ODIzNWRkYTU5ZjM5N2UxMTliOWUxNGY1ZDQzZDEyMTZkYWE0MWFiZGRhNzUyNjQ1YTVjYTIyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.w6lywkkm2EYrSsySt0hWAB5nN2dv70-HBLOAVtDC_XE)

Hi @saifion33,

I have apply your solution but still not worked.

I don't know what to do to solve it.

![Screenshot 2025-02-08 183345](https://github.com/user-attachments/assets/93ab31d7-2b6f-48f9-b11e-b3547b33a53f)

**Comment by benibangun** [comment]
Hi All,

I have found the solution. 

I put this code on  create-form.tsx:

![image](https://github.com/user-attachments/assets/bb31ef0a-8907-4ba1-bcd7-dd7ff3ab779a)
 I also adjust for status.

Problem solved.
